hal/src/main/native/athena/Notifier.cpp

```diff
@@ -161,8 +168,15 @@ HAL_NotifierHandle HAL_InitializeNotifier(int32_t* status) {
 
 HAL_Bool HAL_SetNotifierThreadPriority(HAL_Bool realTime, int32_t priority,
                                        int32_t* status) {
-  auto native = notifierThread.native_handle();
-  return HAL_SetThreadPriority(&native, realTime, priority, status);
+  std::scoped_lock lock(notifierMutex);
+  notifierThreadRealTime = realTime;
+  notifierThreadPriority = priority;
+  if (notifierThread.joinable()) {
+    auto native = notifierThread.native_handle();
+    return HAL_SetThreadPriority(&native, realTime, priority, status);
```

Comment:
Note you're holding the notifierMutex over the call to HAL_SetThreadPriority.  This should be okay, but arguably you could release the lock before making this call.

Reply:
I think we need to hold the lock here, otherwise we could race with notifier close.

Reply:
I think we need to hold it so we don't race with the last close.